src/common/transformations/src/transformations/common_optimizations/gru_cell_fusion.cpp

```diff
@@ -148,6 +148,15 @@ ov::pass::GRUCellFusion::GRUCellFusion() {
             Bh = rg.make<ov::op::v0::Constant>(WRh.get_element_type(), Shape{1, static_cast<size_t>(hidden_size)}, 0);
         }
 
+        // perform additional check for applicability of the transformation
+        // without this check, process_weights can fail
+        if (WR.get_partial_shape()[1] != (hidden_size + input_size)) {
```

Comment:
Can shape size be less than 2 here? Should we add check?

Reply:
It is preliminary checked in the pattern definition. Please see here: https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/gru_cell_fusion.cpp#L91